src/app/search/page.tsx

```diff
@@ -1,17 +1,153 @@
+  };
+}) {
+  const hasActiveFilters = Object.entries(filters).some(([key, value]) => {
+    if (Array.isArray(value)) {
+      return value.length > 0;
+    }
+    return value !== '';
+  });
+  if (!hasActiveFilters) return null;
 
-export default async function (props: { searchParams: Promise<Record<string, string>> }) {
-  const searchParams = await props.searchParams;
-  const filters = Object.fromEntries(
-    Object.entries(searchParams).filter(
-      ([key]) =>
-        ['startTime', 'endTime', 'fromAddress', 'toAddress'].includes(key) ||
-        key.startsWith('term_')
-    )
+  return (
+    <Flex
+      rowGap={2}
+      columnGap={2.5}
+      flexWrap={'wrap'}
+      alignItems={'center'}
+      borderBottom={'1px solid'}
+      borderColor={'borderPrimary'}
+      pb={4}
+    >
+      <Text textStyle={'text-regular-sm'} color={'textSecondary'}>
+        Search results for:
+      </Text>
+      {Object.entries(filters).map(([key, value]) => {
+        if (!value || (Array.isArray(value) && value.length === 0)) return null;
+        const keyword = getKeywordByFilter(key);
+        const displayValue = Array.isArray(value) ? value.join(', ') : value;
+        return (
+          <Text key={key} textStyle={'text-regular-sm'} color={'textPrimary'}>
+            {!!keyword && (
+              <Text
+                textStyle={'text-medium-sm'}
+                fontWeight={'medium'}
+                display="inline-block"
+                bg={{ base: 'feedback.blue-200', _dark: 'feedback.blue-900' }}
+                color={{ base: 'feedback.blue-600', _dark: 'feedback.blue-300' }}
+                borderRadius="redesign.xs"
+                py={1}
+                whiteSpace="pre"
+                textTransform={'uppercase'}
+              >
+                {' '}
+                {keyword}{' '}
+              </Text>
+            )}{' '}
+            {displayValue && filterToFormattedValueMap[key]
+              ? filterToFormattedValueMap[key](displayValue)
+              : displayValue || ''}
+          </Text>
+        );
+      })}
+    </Flex>
   );
+}
+
+function SearchContent({ onTotalChange }: { onTotalChange: (total: number) => void }) {
+  return (
+    <Stack gap={5} mt={5}>
+      <TxTableFilters />
+      <TxSearchResult onTotalChange={onTotalChange} />
+      <TxTableFiltersModal />
+    </Stack>
+  );
+}
 
-  return <Page filters={filters} />;
+function SearchPageContent() {
+  const { fromAddress, toAddress, startTime, endTime, transactionType } = useTxTableFilters();
+  const [resultCount, setResultCount] = useState<number | null>(null);
+
+  const filters = {
+    fromAddress,
+    toAddress,
+    startTime,
+    endTime,
+    transactionType,
+  };
+
+  return (
+    <Stack gap={24} fontFamily="var(--font-instrument-sans)">
+      <Stack gap={3}>
+        <SearchHeader resultCount={resultCount} />
+        <SearchFiltersDisplay filters={filters} />
+        <SearchContent onTotalChange={setResultCount} />
```

Comment:
I would just put the jsx for this component directly into searchpagecontent just because it's just a pass through component and searchcontent and searchpagecontent are too similar that it's confusing